tests/typevar.pyi

```diff
@@ -17,3 +18,41 @@ _TypeVarUsedInBinOp = TypeVar("_TypeVarUsedInBinOp", bound=str)
 
 def func2(arg: _TypeVarUsedInBinOp | int) -> _TypeVarUsedInBinOp | int:
     ...
+
+
+_S = TypeVar("_S")
+
+
+class BadClass:
+    def __new__(cls: type[_S], *args: str, **kwargs: int) -> _S:  # Y019 Use `_typeshed.Self` instead of `_S`, e.g. `def __new__(cls: type[Self], *args: str, **kwargs: int) -> Self: ...`
+        ...
+
+    def bad_instance_method(self: _S, arg: bytes) -> _S:  # Y019 Use `_typeshed.Self` instead of `_S`, e.g. `def bad_instance_method(self: Self, arg: bytes) -> Self: ...`
+        ...
+
+    @classmethod
+    def bad_class_method(cls: type[_S], arg: int) -> _S:  # Y019 Use `_typeshed.Self` instead of `_S`, e.g. `def bad_class_method(cls: type[Self], arg: int) -> Self: ...`
+        ...
+
+
+class GoodClass:
+    def __new__(cls: type[Self], *args: list[int], **kwargs: set[str]) -> Self:
+        ...
+
+    def good_instance_method_1(self: Self, arg: bytes) -> Self:
+        ...
+
+    def good_instance_method_2(self, arg1: _S, arg2: _S) -> _S:
+        ...
```

Comment:
These would be more readable with one-line methods, similar to what typeshed's black configuration produces.

Reply:
Unfortunately that produces errors from stock flake8 because it wants the `...` on its own line. We can fix that but it's probably better done in a separate PR.

Reply:
Yes, that's why I did it the way I did -- the tests didn't pass on my machine if I had them as one-liner functions.